src/components/SelectWithDots/index.tsx

```diff
@@ -88,24 +97,75 @@ export const SelectWithDots: React.ForwardRefExoticComponent<
       }
     }, [defaultValue])
 
-    const CustomMenu = useCallback(({ innerRef, innerProps, children }) => {
-      return (
-        <div
-          ref={innerRef}
-          style={{
-            zIndex: 20,
-            position: 'absolute',
-            top: 35,
-            right: 0,
-            width: 250,
-            paddingBottom: 0,
-          }}
-          {...innerProps}
-        >
-          {children}
-        </div>
-      )
-    }, [])
+    const selectRef = useRef<ReactSelectElement | undefined>(undefined)
+
+    const setSelectRef = useCallback((selectElement: ReactSelectElement) => {
+      if (typeof ref === 'function') {
+        ref(selectElement)
+      } else if (ref != null) {
+        ref.current = selectElement
+      }
+
+      selectRef.current = selectElement
+    }, [ref, selectRef])
+
+    useEffect(() => {
+      if (!open) {
+        selectRef.current?.blur?.()
+      }
+    }, [open, selectRef])
+
+    const CustomMenu = useCallback(
+      ({ innerRef, innerProps, children }) => {
+        let overrides: React.CSSProperties = {}
+        switch (menuAlignment) {
+          case 'bottom-left':
```

Comment:
I personally like using object literals instead of switch cases.

```
const alignmentStyleObj = {
  bottom-left: {
     top: 35,
     left: 0,
  }
}

return alignmentStyleObj[menuAlignment]
```

What do you think?

Reply:
I can _switch_ to that if it's preferred (no pun intended). what I like about `switch`es is being able to share logic with fallthrough cases and not paying the cost of the branches you don't use, e.g.
```tsx
const alignmentStyleOverrides: Record<typeof menuAlignment, React.CSSProperties> = {
  "bottom-left": {
      top: runExpensiveFunction(), // always runs
...
```